### ISSUE TITLE
Major feature request: Remove this title with a descriptive title.

### ISSUE THREAD

**Issue by zasexton** [SimVascular/svFSIplus#238]
### Use Case

There is a need to include symbolic manipulation into `svFSIplus` for custom implementations of the advection-reaction-diffusion equation (and potentially future constitutive models). See https://github.com/SimVascular/svFSIplus/issues/218#issue-2384590596 

### Problem

Figuring out how to include a symbolic library in a way that is efficient and not cumbersome to the rest of the `svFSIplus` project will be important. Many such packages can be large, involving many other dependencies -- we do not want this. 

### Solution

We plan to optionally adopt `SymEngine`, which is a standalone fast C++ symbolic manipulation library, for the symbolic manipulation of custom reactive terms that may be explored in custom implementations of the advection-diffusion-reaction equation. We plan to include this library as an optional Fetch dependency during configuration using cmake's `FetchContent` functionality. This might also outline a good template for interfacing other packages with `svFSIplus` without having to explicitly include them in the code base. 

### Alternatives considered

None so far....other symbolic libraries are typically larger and more difficult to integrate. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct and Contributing Guidelines

**Comment by zasexton** [comment]
an initial implementation has been built that correctly compiles `svFSIplus` with `symengine` here: https://github.com/zasexton/svFSIplus_public/tree/add-symengine-%23238

**Comment by alisonmarsden** [comment]
I think this is now part of the refactoring effort that @zasexton is working on.   Maybe we can create an issue for that instead?   @ktbolt thoughts?

**Comment by ktbolt** [comment]
@alisonmarsden It's not clear to me what exactly `symengine` does and why we need it.

**Comment by zasexton** [comment]
The refactoring that we are doing now manages to avoid a dependence on `symengine` and instead the differentiation machinery now exists natively in the refactored machinery although it relies on the same ideas that i originally explored when creating this issue/branch.

**Comment by alisonmarsden** [comment]
Ok, so seems like we can close this issue then?

**Comment by zasexton** [comment]
yes, the machinery has already been migrated without the extra `symengine` dependency. Will close this out now
